general-superstaq/general_superstaq/service.py

```diff
@@ -318,6 +318,15 @@ def submit_aces(
                 - lifespan: How long to store the jobs submitted for in days (only works with right
                 permissions).
                 - method: Which type of method to execute the circuits with.
+                - noise: A tuple indicating a noise model to simulate the run with. The tuple will
+                    have the form ("channel_name", prob_of_error). The permitted channels are:
```

Comment:
* imo the noise model and error probability should be two separate arguments, instead of having to be passed as a tuple
* `cirq.NoiseModel`s are json-serializable - would it make sense to allow them to be passed as well?

also - predates this pr, but all the parameters we're documenting here they should really be made into optional arguments instead of using kwargs

Reply:
> cirq.NoiseModels are json-serializable - would it make sense to allow them to be passed as well?

Makes sense. This will have to be hacky on the qiskit side though.

Reply:
> imo the noise model and error probability should be two separate arguments, instead of having to be passed as a tuple

Sounds good.

> cirq.NoiseModels are json-serializable - would it make sense to allow them to be passed as well?

I was thinking about doing this, but I don't think there's an exact equivalence between Qiskit's and Cirq's models, so it isn't as easy to convert back and forth. Or we could allow `cirq.NoiseModel` coming from css and keep it as it is for qss.

> predates this pr, but all the parameters we're documenting here they should really be made into optional arguments instead of using kwargs

I'll change them to optional parameters.

Reply:
> Or we could allow cirq.NoiseModel coming from css and keep it as it is for qss.

Can we not use `NoiseModel.to_dict()` and `from_dict()` for qss?

Reply:
> Can we not use NoiseModel.to_dict() and from_dict() for qss?

Ah yeah, I think this could work.

Reply:
For reference, I think the best way to translate from Qiskit's to Cirq's noise models is to use [`Kraus(qiskit_model)`](https://qiskit.org/documentation/stubs/qiskit.quantum_info.Kraus.html) and then use [`cirq.KrausChannel`](https://quantumai.google/reference/python/cirq/KrausChannel) to go back and forth. Let me know if you think there is a better way to do this.

Reply:
> (even tag and lifespan? these two are also optional when creating jobs)

imo they should be optional arguments in `create_job` too, if they are intended for external use - if nothing else it's a form of documentation (part of #523)

Reply:
> imo they should be optional arguments in create_job too

sounds good! I'll make the change for this function for now.

Reply:
> use [`cirq.KrausChannel`](https://quantumai.google/reference/python/cirq/KrausChannel) to go back and forth

`qss` shouldn't be given a `cirq` dependency, unfortunately

i kinda think it's okay for parity to be imperfect here though (or if nothing else it seems preferable to keeping a useful functionality out of one or the other client just for the sake of parity). We'll still be able to specify a noise model in `qss` using the string method. I imagine that we'll also be making our error models more general/customizable, which would hopefully expand this to cover a lot more use cases

we can also add an issue to revisit this in the future

Reply:
> qss shouldn't be given a cirq dependency, unfortunately

Oh, I meant server-side, sorry I wasn't specific. In the client I think it's possible to let the users send a serialized `cirq.NoiseModel` and a `qiskit_aer.noise.QuantumError`.

Reply:
It looks like [`qiskit_aer.noise.NoiseModel.from_dict`](https://qiskit.org/ecosystem/aer/_modules/qiskit_aer/noise/noise_model.html#NoiseModel.from_dict) is being deprecated, and I don't think there's another way to reconstruct a Qiskit noise model back in the server.